### DISCUSSION TITLE
[xy-chart] Horizontal bar

### DISCUSSION THREAD

**Discussion by maximerivolet** [airbnb/visx#1050] [Q&A] [answered]
Hi,

I am trying to use the XYchart to do an horizontal barchart, like this one :

![image](https://user-images.githubusercontent.com/12681339/107124229-d4da5080-68a2-11eb-9897-708f68fb28cc.png)

Do you have a working example of that ?

Regards,

**Comment by williaster** [comment, marked as answer]
Hey @goldenivan 👋   looks like you found the `xychart` demo on the demo site [here](https://airbnb.io/visx/xychart). You can view the source code for any example below its rendering, and also follow the [codesandbox link](https://codesandbox.io/s/github/airbnb/visx/tree/master/packages/visx-demo/src/sandboxes/visx-xychart) displayed on the example page to tinker live.

In general `XYChart` will try to infer whether you are rendering items horizontally based on your x/y scale types but you can also pass `horizontal={true}` as described [in the docs](https://airbnb.io/visx/docs/xychart#XYChart_horizontal). Here all we are doing to toggle vertical / horizontal is 
- [switching which config we pass as `xScale` vs `yScale`](https://github.com/airbnb/visx/blob/master/packages/visx-demo/src/sandboxes/visx-xychart/ExampleControls.tsx#L190)
- [switching x/y accessor functions on the series](https://github.com/airbnb/visx/blob/master/packages/visx-demo/src/sandboxes/visx-xychart/ExampleControls.tsx#L171)

Hope that helps.

**Reply by williaster** [reply]
Quick note that looks like the code sandbox link is broken, which is due to the demo site using `@visx/xychart` features that are not yet published on `npm`. I'll try to get the `1.5.0` out soon and that should fix itself. For now you can use [this link]([url](https://codesandbox.io/s/keen-goldstine-j3nnh)) which removes the unreleased `AreaStack` component.

**Reply by maximerivolet** [reply]
Hi @williaster, thanks for the answer. I saw the codesandbox link. But as you said, it is currently broken. Currently, I'm picking parts of the code I try to reproduce the "BarSeries" example.

You mentionned "this link", but you forgot it, didn't you ?

I understand that you switch the x/y scale based on a boolean. But currently, I'm failing in displaying a simple XYChart, based on the examples I saw.

**Reply by williaster** [reply]
Oh snap, sorry [here's the link](https://codesandbox.io/s/red-firefly-x1fb4?file=/ExampleControls.tsx) 🤦 

That demo is kind of "the whole kitchen sink," so if you're looking for a more simple example to get started you can check out [this one](https://codesandbox.io/s/themeprovider-sbdvz?file=/Example.tsx) which is linked from the [`/docs/xychart`](https://airbnb.io/visx/docs/xychart) advanced usage examples (it is demoing a custom theme, but is a basic bar chart).

**Reply by williaster** [reply]
And here's that simple demo [updated for horizontal rendering](https://codesandbox.io/s/themeprovider-forked-qldbf?file=/Example.tsx) (just swapped x/yScale config + accessors).

<img src="https://user-images.githubusercontent.com/4496521/107565645-bcb34b80-6b98-11eb-871d-0e72111312e2.png" width="500" />

**Comment by maximerivolet** [comment]
I was able to do that for my project. Thank you for your time !

**Reply by VigneshTrantor** [reply]
@goldenivan I was able to do it by flipping the XY chart xScale and yScale values "{ type: "band" }" and "{ type: "linear" }". Just interchange them.

Also, you will need to flip the accessors
`export const DATUM_ACCESSORS = (isVertical: boolean) => ({
  xAccessor: (d: any) => (isVertical ? d.x : d.y),
  yAccessor: (d: any) => (isVertical ? d.y : d.x),
});`

We had a requirement to be able to flip the chart on clicking of the radio button.
